src/org/rascalmpl/parser/gtd/SGTDBF.java

```diff
@@ -1134,6 +1264,11 @@ protected AbstractNode parse(AbstractStackNode<P> startNode, URI inputURI, int[]
 	    this.recoverer = recoverer;
 	    this.debugListener = debugListener;
 
+		if (inputURI != null) {
+			String query = inputURI.getQuery();
+			visualizer = query != null && query.contains("visualize=true") ? new ParseStateVisualizer("Parser") : null;
```

Comment:
This check should indeed include a check for `VISUALIZATION_ENABLED`.

The alternative to this approach is adding an extra keyword parameter but at this stage I did not want to extend the public parser API with a debug-only parameter. Maybe someday visualization is good enough to be core parser feature.

Also, I have been pestering Davy with the request to add a "ParserConfig" object instead of a growing list of keyword parameters to the parser functions so adding a new element is no longer a hassle, but so far he is not convinced.

Reply:
Since my name was mentioned: Yes, that might be nice to replace all those arguments, but in a way, a parser config would again have keyword arguments (or some other form), so it will still be roughly as verbose?

If you come up with a backwards compatible design proposal, lets talk about it. But that doesn't have to be part of this PR I would guess.